marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/EnvironmentController.java

```diff
@@ -76,6 +81,11 @@ public SystemInfo getSystemInfo() {
         info.marklogicVersion = versionInfo.getMarkLogicVersion();
         info.host = hubCentral.getHost();
         info.sessionTimeout = environment.getProperty("server.servlet.session.timeout");
+        Optional.of(session).ifPresent((session) -> {
```

Comment:
Autowiring in the HttpSession makes me nervous - I think it's simpler just to declare an HttpServletRequest as a parameter of getSystemInfo() (you may be able to declare HttpSession as a parameter too, I forget if Spring MVC will populate it automatically). You can get the session from the HttpServletRequest.

Reply:
Just verified, you can just declare HttpSession as a parameter - https://www.logicbig.com/tutorials/spring-framework/spring-web-mvc/http-session-param.html - I think that's the better way to go, also makes testing easier. Note that the session will never be null too.

Reply:
Moved to a parameter.